WebAPI/Middleware/ExceptionMiddleware.cs

```diff
@@ -44,33 +45,37 @@ public async Task InvokeAsync(HttpContext context)
                         Values = errorMessages.Distinct().ToArray()
                     })
                 .ToDictionary(x => x.Key, x => x.Values);
-            context.Response.ContentType = "application/json";
-            context.Response.StatusCode = (int)HttpStatusCode.UnprocessableEntity;
-
-            var response = new
-                ErrorDetails(context.Response.StatusCode, e.Message, "Validation Error") { ValidationErrors = validationErrors };
-            
-            var json = JsonSerializer.Serialize(response,
-                new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase, WriteIndented = true });
 
-            await context.Response.WriteAsync(json);
+            await GenerateExceptionResponse(context, (int)HttpStatusCode.UnprocessableEntity, e, "Validation error", validationErrors);
+        }
+        catch (EntityNotFoundException e)
+        {
+            await GenerateExceptionResponse(context, (int)HttpStatusCode.NotFound, e, "No entity with the given parameter was found");
         }
         catch (Exception e)
         {
             this._logger.LogError(e, e.Message);
 
-            context.Response.ContentType = "application/json";
-            context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+            await GenerateExceptionResponse(context, (int)HttpStatusCode.InternalServerError, e, "Something went wrong");
+        }
+    }
 
-            var response = this._env.IsDevelopment()
-                ? new ErrorDetails(context.Response.StatusCode, e.Message, e.StackTrace)
-                : new ErrorDetails(context.Response.StatusCode, e.Message, "Something went wrong");
-            
-            var opt = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+    private async Task GenerateExceptionResponse(HttpContext context, int statusCode, Exception e, string details, [Optional] Dictionary<string, string[]> validationErrors)
+    {
+        context.Response.ContentType = "application/json";
+        context.Response.StatusCode = statusCode;
 
-            var json = JsonSerializer.Serialize(response, opt);
+        var response = new ErrorDetails(context.Response.StatusCode, e.Message, details);
 
-            await context.Response.WriteAsync(json);
+        if (validationErrors is not null)
+        {
+            response.ValidationErrors = validationErrors;
         }
+
+        var opt = new JsonSerializerOptions { PropertyNamingPolicy = JsonNamingPolicy.CamelCase };
+
+        var json = JsonSerializer.Serialize(response, opt);
+
+        await context.Response.WriteAsync(json);
```

Comment:
Did you test it out? Still works like before?

Reply:
I tested it, it works.